Main.elm

```diff
@@ -185,25 +221,27 @@ padView model =
                 "pad"
         ]
         [ controlBar model
-        , Html.div
-            [ Html.Attributes.class <|
-                case ( model.stored, model.reveal ) of
-                    ( True, True ) ->
-                        "stored reveal"
-
-                    ( True, False ) ->
-                        "stored"
-
-                    ( False, True ) ->
-                        "reveal"
-
-                    ( _, _ ) ->
-                        ""
-            , Html.Attributes.contenteditable True
-            , Html.Attributes.property "innerHTML" (Json.Encode.string model.content)
-            ]
-            []
+        , contentEditable model
+        ]
+
+
+innerHtmlDecoder =
+    Json.Decode.at [ "target", "innerHTML" ] Json.Decode.string
+
+
+contentEditable : Model -> Html.Html Msg
+contentEditable model =
+    Html.div
+        [ Html.Attributes.class <|
+            if model.reveal then
+                "reveal"
+            else
+                ""
```

Comment:
I remember you talking about extracting the `contentEditable` from the `stored` green flash, but I forget what the exact reasoning was. Was it because doing the green flash redrew this element, which meant resetting `innerHTML` (and thus bouncing the cursor)? If so, why are we OK leaving `reveal` here? Is it because to toggle `reveal` you have to click the mouse, so we figure it's OK if the cursor bounces when that happens?